### PR TITLE
feat: Create plugin alias filter to use category and or site

### PR DESCRIPTION
**Pull request by Aiky30**
* feat: Clean out old sass, js and css dist from gulp and package.json
* feat: Create plugin alias filter to use category and or site
* fix: Outdated build scripts from Node 5 to Node 16

Includes cleanup and updating the build system. I would have liked to have done the cleanup in a separate change but due to the recent changes being coupled to the rework it was difficult to justify the additional effort to split the changes. This goes against my own principles where changes should be kept separate so I have not taken this decision lightly.

### CODE REVIEW

**Review by Bernardvdv — APPROVED** [review]

File: tests/test_views.py [hunk 1]
```diff
@@ -1090,6 +1092,59 @@ def test_select2_view_category_and_term(self):
             [alias1.pk],
         )
 
+    def test_select2_view_site(self):
+        """
+        The list should be filtered based on only matching
```
Comment (adam-murray) [inline tests/test_views.py:1097]:
```suggestion
        The list is filtered based on only matching
```

File: tests/test_views.py [hunk 2]
```diff
@@ -1090,6 +1092,59 @@ def test_select2_view_category_and_term(self):
...
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            [a['id'] for a in response.json()['results']],
+            [alias1.pk, alias2.pk],
+        )
+
+    def test_select2_view_site_and_category(self):
+        """
+        The list should be filtered based on only matching
```
Comment (adam-murray) [inline tests/test_views.py:1121]:
```suggestion
        The list is filtered based on only matching
```

File: tests/test_views.py [hunk 3]
```diff
@@ -1502,6 +1557,164 @@ def test_alias_multisite_support(self):
...
+                data={'limit': 2},
+            )
+
+        content = response.json()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(content['more'])
+        self.assertEqual(len(content['results']), 2)
+
+    def test_select2_view_text_repr(self):
```
Comment (adam-murray) [inline tests/test_views.py:1647]: Missing docstring

**Review by adam-murray — CHANGES_REQUESTED** [review]
Looks good, jsut a couple of minor changes to docstrings in the tests

**Review by NarenderRajuB — APPROVED** [review]
LGTM

File: tests/test_views.py (hunk 3 above)
Comment (Aiky30) [inline tests/test_views.py:1647]:
```suggestion
    def test_select2_view_text_repr(self):
        """
        Ensure that the display / text representation of the object
        is output to the user.
        """
```

**Review by adam-murray — APPROVED** [review]

**Pull request merged**